how-to/customize-workspace/client/src/logger-shapes.ts

```diff
@@ -0,0 +1,69 @@
+export interface Logger<T = unknown> {
+	/**
+	 * Optionally initialize the logger.
+	 * @param options The custom options for the logger.
+	 */
+	initialize?(options: T): Promise<void>;
+
+	/**
+	 * Optionally close down the logger.
+	 */
+	closedown?(): Promise<void>;
+
+	/**
+	 * Log data as information.
+	 * @param group The group sending the log message.
+	 * @param message The message to log.
+	 * @param optionalParams Optional parameters for details.
+	 */
+	info(group: string, message: unknown, ...optionalParams: unknown[]): void;
```

Comment:
I think we should also include host i.e. the identity of the window/view that is doing the logging that way when we have more windows (we could expose a channel and a tiny logger js for other windows that communicates via the channel) they can log as well and we can see where the error was thrown.

Reply:
It could be defaulted to the fin.me.identity.name in the logger itself so that all calls to the logger across ts files don't have to retype that (and the cut down channel logger for other windows can do that call).